src/main/java/com/minecolonies/coremod/util/ItemStackUtils.java

```diff
@@ -0,0 +1,391 @@
+     * @return true if tool is acceptable
+     */
+    public static boolean verifyToolLevel(@NotNull final ItemStack itemStack, final int toolLevel, final int minimalLevel, final int maximumLevel)
+    {
+        if (toolLevel < minimalLevel)
+        {
+            return false;
+        }
+        return (toolLevel + getMaxEnchantmentLevel(itemStack) <= maximumLevel);
+    }
+
+    /**
+     * Calculates the fortune level this tool has.
+     *
+     * @param tool the tool to check.
+     * @return fortune level.
+     */
+    public static int getFortuneOf(@Nullable final ItemStack tool)
+    {
+        if (tool == null)
+        {
+            return 0;
+        }
+        //calculate fortune enchantment
+        int fortune = 0;
+        if (tool.isItemEnchanted())
+        {
+            final NBTTagList t = tool.getEnchantmentTagList();
+
+            for (int i = 0; i < t.tagCount(); i++)
+            {
+                final int id = t.getCompoundTagAt(i).getShort(NBT_TAG_ENCHANT_ID);
+                if (id == FORTUNE_ENCHANT_ID)
+                {
+                    fortune = t.getCompoundTagAt(i).getShort(NBT_TAG_ENCHANT_LEVEL);
+                }
+            }
+        }
+        return fortune;
+    }
+
+    /**
+     * Calculates the max level enchantment this tool has.
+     *
+     * @param itemStack the tool to check.
+     * @return max enchantment level.
+     */
+    public static int getMaxEnchantmentLevel(final ItemStack itemStack)
+    {
+        if (itemStack == null)
+        {
+            return 0;
+        }
+        int maxLevel = 0;
+        if (itemStack != null)
+        {
+            final NBTTagList nbttaglist = itemStack.getEnchantmentTagList();
+
+            if (nbttaglist != null)
+            {
+                for (int j = 0; j < nbttaglist.tagCount(); ++j)
+                {
+                    final int level = nbttaglist.getCompoundTagAt(j).getShort("lvl");
+                    maxLevel = level > maxLevel ? level : maxLevel;
+                }
+            }
+        }
+        return maxLevel;
+    }
+
+
+    /**
+     * Checks if an item serves as a weapon.
+     *
+     * @param stack the stack to analyze.
+     * @return true if it is a tool or sword.
+     */
+    public static boolean doesItemServeAsWeapon(@NotNull final ItemStack stack)
+    {
+        return stack.getItem() instanceof ItemSword || stack.getItem() instanceof ItemTool;
+    }
+
+
+    /**
+     * Assigns a string containing the grade of the toolGrade.
+     *
+     * @param toolGrade the number of the grade of a tool
+     * @return a string corresponding to the tool
+     */
+    public static String swapToolGrade(final int toolGrade)
+    {
+        switch (toolGrade)
+        {
+            case 0:
+                return "Wood or Gold";
+            case 1:
+                return "Stone";
+            case 2:
+                return "Iron";
+            case 3:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)